src/PathautoManager.php

```diff
@@ -122,15 +128,31 @@ public function __construct(ConfigFactoryInterface $config_factory, ModuleHandle
   /**
    * {@inheritdoc}
    */
-  public function createAlias($module, $op, $source, $data, $type = NULL, $langcode = LanguageInterface::LANGCODE_NOT_SPECIFIED) {
-    $config = $this->configFactory->get('pathauto.settings');
-
+  public function createAlias(EntityInterface $entity, $op) {
     // Retrieve and apply the pattern for this content type.
-    $pattern = $this->getPatternByEntity($module, $type, $langcode);
+    $pattern = $this->getPatternByEntity($entity);
+    if (empty($pattern)) {
+      // No pattern? Do nothing (otherwise we may blow away existing aliases...)
+      return NULL;
+    }
+
+    $entity_type_id = $entity->getEntityTypeId();
+    $source = $entity->urlInfo()->toString();
```

Comment:
Oh, almost forgot. The current code uses '/' . $entity->urlInfo()->getInternalPath(). This difference is important. getInternalPath() was deprecated for a long time, but there is no replacement, so it no longer is.

You can't see a difference when you're not in a subdirectory, but to String() includes that while getInternalPath() does not, assuming a checkout subdirectory like testbot:
toString: /checkout/node/1 (also already includes the alias, unless you explicitly opt-out)
getInternalPath: node/1.